### PR TITLE
Update to python scripts

### PR DESCRIPTION
**Pull request by rettinghaus**
This PR updates the scripts for building source files for different languages to Python 3 and does some minor formatting.

It merges back handling of data types and data lists from Verovio branch.

### CODE REVIEW

File: tools/parseschema2.py [hunk 1]
```diff
@@ -271,18 +275,18 @@ def geteldocs(self, ename):
...
-        if os.path.exists(output_directory):
-            lg.debug("Removing old Java output directory")
-            shutil.rmtree(output_directory)
-        os.mkdir(output_directory)
-        java.create(schema, output_directory)
-        if args.includes:
-            java.parse_includes(output_directory, args.includes)
-
-    if "csharp" in l_langs.lower():
+        if "java" in l_langs.lower():
```
Comment (apacha) [inline tools/parseschema2.py]: Is this indentation intended?

Comment (rettinghaus) [reply tools/parseschema2.py]: Yes, in order to treat every language the same.

File: tools/langs/java.py [hunk 2]
```diff
@@ -84,20 +84,19 @@ def __create_java_classes(schema, outdir):
...
             fmi.close()
             lg.debug("\tCreated {0}".format(file_name, class_name))
 
 
 def __parse_codefile(methods, includes, directory, codefile):
-    f = open(os.path.join(directory, codefile), 'r')
-    contents = f.readlines()
-    f.close()
+    f = Path(directory, codefile)
+    f.open()
```
Comment (apacha) [inline tools/langs/java.py]: Why are you opening the file here?

**Comment by rettinghaus** [comment]
@apacha please have another look at the code and merge, so we may proceed updating LibMEI itself.

File: tools/langs/python.py [hunk 3]
```diff
@@ -54,106 +55,101 @@ def create(schema, outdir):
...
     return (ret, inc)
 
+
 def __parse_codefile(methods, includes, directory, codefile):
-    f = open(os.path.join(directory, codefile), 'r')
-    contents = f.readlines()
-    f.close()
-    regmatch = re.compile(r"[\s]+# <(?P<elementName>[^>]+)>", re.MULTILINE|re.DOTALL)
+    f = Path(directory, codefile)
+    contents = f.read_text()
```
Comment (apacha) [inline tools/langs/python.py:138]: The signature here has changed and is now incompatible (`contents` used to be `List[str]` and now is `str`). Either use `f.read_text().splitlines()` or use another method to read the file line by line.

File: tools/langs/python.py [hunk 4]
```diff
@@ -54,106 +55,101 @@ def create(schema, outdir):
...
-
-
-
-
-
+            if match.group("elementName") in list(methods.keys()):
+                contents[i] = methods[match.group(
+                    "elementName")].lstrip("\n") + "\n"
 
+    f.write_text(contents)
```
Comment (apacha) [inline tools/langs/python.py:154]: If you use `.splitlines()` you will have to update this call as well, e.g., `f.write_text(os.linesep.join(contents))`

File: tools/langs/csharp.py [hunk 5]
```diff
@@ -258,7 +268,7 @@ def __create_att_classes(schema, outdir):
 
     outdir_att = os.path.join(outdir, "atts")
```
Comment (apacha) [inline tools/langs/csharp.py]: Found an old os.path.join

File: tools/langs/cplusplus.py [hunk 6]
```diff
@@ -231,13 +233,13 @@ def __create_mixin_classes(schema, outdir):
...
             'license': LICENSE.format(authors=AUTHORS),
-            'moduleNameCaps': "{0}MIXIN".format((module.upper()).replace("-","_")),
+            'moduleNameCaps': "{0}MIXIN".format((module.upper()).replace("-", "_")),
             'elements': classes.strip()
         }
         if "std::string" in classes:
             tplvars["includes"] = "#include <string>"
         fullout = CLASSES_HEAD_TEMPLATE.format(**tplvars)
-        fmh = open(os.path.join(outdir, "{0}mixins.h".format(module.lower())), 'w')
+        fmh = open(Path(outdir, "{0}mixins.h".format(module.lower())), 'w')
```
Comment (apacha) [inline tools/langs/cplusplus.py]: Why not fully changing it to
```python
fmh = Path(outdir, "{0}mixins.h".format(module.lower())) # whatever fmh stands for
fmh.write_text(fullout)
```

File: tools/langs/cplusplus.py [hunk 7]
```diff
@@ -473,19 +481,20 @@ def __parse_includefile(contents):
...
     inc = re.findall(reginc, contents)
     return (ret, inc)
 
 
 def __parse_codefile(methods, includes, directory, codefile):
-    f = open(os.path.join(directory, codefile), 'r')
-    contents = f.readlines()
-    f.close()
+    f = Path(directory, codefile)
+    contents = f.read_text()
```
Comment (apacha) [inline tools/langs/cplusplus.py:491]: Same signature mismatch.

File: tools/langs/cplusplus_vrv.py [hunk 8]
```diff
@@ -882,32 +917,27 @@ def __create_att_classes(schema, outdir, includes_dir):
...
 
 def __process_include(fname, includes, includes_dir):
-    name,ext = os.path.splitext(fname)
     new_methods, includes_block = None, None
     if "{0}.inc".format(fname) in includes:
         lg.debug("\tProcessing include for {0}".format(fname))
-        f = open(os.path.join(includes_dir, "{0}.inc".format(fname)), 'r')
-        includefile = f.read()
-        f.close()
+        f = Path(includes_dir, "{0}.inc".format(fname))
```
Comment (apacha) [inline tools/langs/cplusplus_vrv.py]: I wouldn't call a path variable `f` as it strongly suggests to contain something like a file. If you never use it again, just make it a one-liner:
```python
includefile = Path(includes_dir, "{0}.inc".format(fname)).read_text()
```

File: tools/langs/cplusplus_vrv.py [hunk 9]
```diff
@@ -926,9 +956,8 @@ def __parse_includefile(contents):
     return (ret, inc)
 
 def __parse_codefile(methods, includes, directory, codefile):
-    f = open(os.path.join(directory, codefile), 'r')
-    contents = f.readlines()
-    f.close()
+    f = Path(directory, codefile)
+    contents = f.read_text()
```
Comment (apacha) [inline tools/langs/cplusplus_vrv.py:958]: Signature mismatch

**Review by apacha — CHANGES_REQUESTED** [review]
I did a basic check of the code changes and left some comments, where I found issues.
I still didn't run the generator or anything, but just read through the code.

Would it make sense to also add a requirements.txt at least for the following two dependencies? 
```
pyyaml
lxml
```

File: tools/langs/csharp.py (hunk 5 above)
Comment (rettinghaus) [reply tools/langs/csharp.py]: Thanks for catching this one!

File: tools/langs/python.py (hunk 3 above)
Comment (rettinghaus) [reply tools/langs/python.py:138]: Thanks for pointing this out. I rechecked again output before/after, and it didn't change. So here we're safe.

File: tools/langs/cplusplus_vrv.py (hunk 8 above)
Comment (rettinghaus) [reply tools/langs/cplusplus_vrv.py]: Thanks! Done.

**Comment by rettinghaus** [comment]
@apacha Thanks for the review! I re-checked parsing all languages again, and everything works fine.

**Pull request merged**